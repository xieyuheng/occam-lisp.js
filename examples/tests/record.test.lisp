[:x 1 :y 2 :z 3]

(assert (equal? [:x 1 :y 2 :z [:x 1 :y 2]]
                [:x 1 :y 2 :z [:x 1 :y 2]]))

(assert (not (equal? [:x 1 :y 2 :z [:x 1 :y 2]]
                     [:x 1 :y 2 :z [:x 1 :y 3]])))

(assert (not (equal? [:x 1 :y 2 :z [:x 1 :y 2]]
                     [:x 1 :y 2 :z [:x 1 :y 2] :w 3])))

(assert (equal? (record-length [1 2 3 4 5 :x 1 :y 2 :z 3])
                3))
