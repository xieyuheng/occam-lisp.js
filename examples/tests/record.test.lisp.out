[:x 1 :y 2 :z 3]
#t
#t
#t
